consensus/src/detail/mod.rs

```diff
@@ -643,4 +641,4 @@ impl<'a> ConsensusRef<'a> {
 }
 
 #[cfg(test)]
-mod test;
+mod tests;
```

Comment:
To get proper unit-tests visibility to all components, even private ones, it should be `test`, not `tests`:

https://doc.rust-lang.org/reference/visibility-and-privacy.html

Reply:
> To get proper unit-tests visibility to all components, even private ones, it should be `test`, not `tests`:
> 
> https://doc.rust-lang.org/reference/visibility-and-privacy.html

I think the name does not matter. Modules always have access to private fields in parent scope, regardless of the name, there is no special case for modules called `test`. The docs just point out that if you add `mod test` as a submodule of code under test, you get access to private fields of stuff in ancestor modules. That is a consequence of the parent-child module relationship, not module name. I am not 100% certain about this but I think I'm right.

Reply:
You basically elaborated my point.

Reply:
@iljakuklic Yes, you are right! Anyway, if `test` looks better, I wouldn't mind renaming it.